REPO-3950: Support permissions in node creation

AFaust opened the pull request:
This PR adds the missing capability to specify permissions during node creation. This has already been filed as a story in JIRA via https://issues.alfresco.com/jira/browse/REPO-3950. I have extracted the permission handling code from the node update handling into a separate method and re-used it in node creation. I have copied & adapted the unit tests for update handling to provide tests for handling the creation case.

Unfortunately, since I could make none of the (original) unit tests work (NPE during test setup), I could not run & verify that my changes work via the unit tests - instead I had to rely on running a patched Alfresco CE 6.1.1 Docker image and doing manual ReST calls.

derekhulley commented:
@AFaust , we have recently put some Travis configuration into master.
If you rebase, can confirm a green build?

AFaust commented:
There seems to be only one issue after rebasing and running Travis - some response code is not what it should have been. I'll have to check the unit test code (that I mostly copied + adapted from existing tests) and see if that can be corrected.

AFaust commented:
@derekhulley Turns out the failing test was the result of another bug uncovered by my code reorganisation (to be able to reuse code). I also fixed that bug andadded a test for the specific update, which was missing before. All tests are green now, so you guys can review for the actual merge.

montgolfiere commented:
I met with @AFaust @ DevCon. It would be great if we could progress the review of this PR :-)

killerboot commented:
It is in our backlog (not too far).
